For_university/experiment1.py

```diff
@@ -0,0 +1,72 @@
+import random
+
+SIZE_OF_UNIVERSE = 100
+NUMBER_OF_SUBSETS = 40
+CARDINALITY = 4
+REMAINING_SUBSET = NUMBER_OF_SUBSETS - CARDINALITY
+ELEMENTS_OF_UNIVERSE = list(range(SIZE_OF_UNIVERSE))
+RATIO = int(NUMBER_OF_SUBSETS / CARDINALITY)
+NUMBER_OF_DUPLICATE = int((NUMBER_OF_SUBSETS-CARDINALITY)/(CARDINALITY/2))
+UNCOVERED_ELEMENTS = set(ELEMENTS_OF_UNIVERSE)
+TEST_LOOP = 1
+
+
+def making_universe(SIZE_OF_UNIVERSE, CARDINALITY):
+    universe = []
+    i = 0
+    small_subset = int(SIZE_OF_UNIVERSE / CARDINALITY - 1)
+    big_subset = int(SIZE_OF_UNIVERSE / CARDINALITY + 1)
+    last_element = 0
+    for index in range(CARDINALITY):
+        if index < int(CARDINALITY / 2):
+            subset = list(ELEMENTS_OF_UNIVERSE[last_element:last_element + big_subset])
+            for i in range(NUMBER_OF_DUPLICATE+1):
+                universe.append(subset)
+            last_element = last_element + big_subset
+        else:
+            universe.append(list(ELEMENTS_OF_UNIVERSE[last_element:last_element + small_subset]))
+            last_element = last_element + small_subset
+
+
+    return universe
+
+
+def coverage_function(subset):
+    return len(set(UNCOVERED_ELEMENTS).difference(set(subset)))
+
+
+def find_best_in_segment(subsets_in_segment):
+    best_subset = (min(subsets_in_segment, key=coverage_function))
+    return best_subset
+
+
+def proposed_algorithem(subsets):
+    global UNCOVERED_ELEMENTS
+    solution = list()
+    for i in range(CARDINALITY):
+        best_marginal_subset = find_best_in_segment(subsets[i * RATIO:i * RATIO + RATIO])
+        solution.append(best_marginal_subset)
+        UNCOVERED_ELEMENTS = UNCOVERED_ELEMENTS.difference(set(best_marginal_subset))
```

Comment:
1. python `set`s "override" the `+-><...`operators so you can write `set_diff = set1 - set2`
2. you assign some calculation to the same variable (`UNCOVERED_ELEMENTS`). that's very confusing. It also means the variable is misnamed (before or after the assignment). I'm not sure why you did that, but try to avoid it!